python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -150,40 +280,71 @@ def _get_tensor_meta(self, tensor: "torch.Tensor") -> Optional["TorchTensorType"
 
     def write(
         self,
-        tensor: "torch.Tensor",
+        tensors: Union["torch.Tensor", List["torch.Tensor"], Exception],
     ):
-        if isinstance(tensor, ray.exceptions.RayTaskError):
+        if isinstance(tensors, ray.exceptions.RayTaskError):
             # TODO(swang): Write exceptions to the meta channel if it is
             # available.
-            raise tensor
-
-        meta = self._get_tensor_meta(tensor)
-        if meta is not None:
-            self._meta_channel.write(meta)
+            raise tensors
+
+        if isinstance(tensors, list):
```

Comment:
nit: make tensors = [tensors] and always use this branch instead?

Reply:
Hmm prefer to keep it this way because sometimes we send a list of metadata and sometimes we send just one.